pkg/virt-handler/non-root.go

```diff
@@ -135,3 +138,58 @@ func (d *VirtualMachineController) nonRootSetup(origVMI, vmi *v1.VirtualMachineI
 	}
 	return nil
 }
+
+func (d *VirtualMachineController) ensurePVCsOwnershipFromDVs(vmi *v1.VirtualMachineInstance) error {
+	res, err := d.podIsolationDetector.Detect(vmi)
+	if err != nil {
+		return err
+	}
+	passthoughFSVolumes := make(map[string]struct{})
+	for i := range vmi.Spec.Domain.Devices.Filesystems {
+		passthoughFSVolumes[vmi.Spec.Domain.Devices.Filesystems[i].Name] = struct{}{}
+	}
+
+	pvcVolume := make(map[string]v1.VolumeStatus)
+	hotplugVolumes := make(map[string]bool)
+	for _, volumeStatus := range vmi.Status.VolumeStatus {
+		if volumeStatus.HotplugVolume != nil {
+			hotplugVolumes[volumeStatus.Name] = true
+		}
+
+		if volumeStatus.PersistentVolumeClaimInfo != nil {
+			pvcVolume[volumeStatus.Name] = volumeStatus
+		}
+	}
+
+	for i := range vmi.Spec.Volumes {
+		volume := vmi.Spec.Volumes[i]
+		if volumeSource := &vmi.Spec.Volumes[i].VolumeSource; volumeSource.DataVolume != nil {
```

Comment:
this is relavant for PersistentVolumeClaim volume sources as well, right?

Reply:
See https://github.com/kubevirt/kubevirt/pull/6445/files/529e94734b09785b78df60be3c9e3aae4d6885f3#r721272473